### ISSUE TITLE
Better handling of touch events

### ISSUE THREAD

**Issue by HarelM** [openlayers/openlayers#9138]
**Is your feature request related to a problem? Please describe.**
When using pitch-zoom along with a custom interaction it very hard to make it work as the user would expect

**Describe the solution you'd like**
I would like to be able to know when writing a custom interaction if the current tap is pitch-zoom or regular tap.
This might be a noob question, but here goes anyway:
I wrote a custom interaction to allow the user to draw routes on the map.
Can be seen here:
https://israelhiking.osm.org.il/beta/
When clicking the pencil icon in the right side.
Code for this interaction can be found here:
https://github.com/IsraelHikingMap/Site/blob/master/IsraelHiking.Web/sources/application/components/intercations/route-edit-route.interaction.ts

This work well for editing and dragging the map, but when pitch-zooming and touching with one finger on the route editable point the point gets dragged instead of the map performing zoom.

I'm not sure how to solve this...
Relevant issue is described here:
https://github.com/IsraelHikingMap/Site/issues/926

Any help will be appreciated.

**Comment by HarelM** [comment]
Thanks for the info! I'll give it a try and let you know if it solves my issue.

**Comment by simonseyock** [comment]
@ahocevar: Nice to know! Now I might want to refactor some stuff :D

Maybe this information should be made a little bit more visible in the documentation. The only thing i find that covers part of this is the documentation of the handleEvent parameter in the Interaction class:
https://openlayers.org/en/latest/apidoc/module-ol_interaction_Interaction-Interaction.html

Other places where I looked:
1. https://openlayers.org/en/latest/apidoc/module-ol_interaction.html - here something of the order of the interactions is said, but nothing about the effect
1. https://openlayers.org/en/latest/apidoc/module-ol_Map-Map.html#addInteraction - for addLayers there is something said about changing the order of layers

**Comment by marcjansen** [comment]
Documentation enhancement PRs gladly accepted...

**Comment by HarelM** [comment]
I have written a custom interaction that handles both click to add a point and drag to move that point so I think `PointerInteraction` does not suite me since it has some logic there of handling down up etc which I think was not what I needed (It was a bit long ago when I did that...).
Nevertheless,  the pointer counter used there can be used in custom interaction like the one I wrote.
But I can't use it without inheriting.
My proposal here is to move the pointer counting logic to a helper which will facilitate it by composition instead of inheritance.
See here:
https://github.com/IsraelHikingMap/Site/commit/a1d22b28d737d7a50e27cd160cbb9c3de44501a9
